src/syrupy/serializers/base.py

```diff
@@ -70,19 +88,24 @@ def create_or_update_snapshot(self, data: "SerializableData", index: int) -> Non
         self.write(data, index=index)
         self.post_write(data, index=index)
 
-    def delete_snapshot(self, snapshot_file: str, snapshot_name: str) -> None:
+    @final
+    def delete_snapshot_from_file(self, snapshot_file: str, snapshot_name: str) -> None:
         """
         Utility method for removing a snapshot from a snapshot file.
         """
-        self.write_snapshot_or_remove_file(snapshot_file, snapshot_name, None)
+        self._write_snapshot_or_remove_file(snapshot_file, snapshot_name, None)
 
     def pre_read(self, index: int = 0) -> None:
         pass
 
+    @final
     def read(self, index: int = 0) -> "SerializableData":
+        """
+        Override _read_snapshot_from_file in subclass to change behaviour
```

Comment:
This method should not be overridden but rather use `get_filepath`, `get_snapshot_name` and the internal `_read_snapshot_from_file` which can be overridden to customise read behaviour